core/src/main/scala/kafka/log/LogConfig.scala

```diff
@@ -232,7 +232,7 @@ object LogConfig {
     }
   }
 
-  private val configDef: LogConfigDef = {
+  val configDef: LogConfigDef = {
```

Comment:
@ijuma The internal project is a LinkedIn wrapper around the open-source kafka. This wrapper  dumps kafka-server configs into ZooKeeper on startup. It's not all the configs but only those that are configurable per topic (think `kafka-topics.sh`). The topic-level configs are essentially `LogConfig`. Moreover, the names of the keys in `LogConfig` and `KafkaConfig` are different. That's the whole purpose of `Logconfig.configDef.serverDefaultConfigNames`. 

Reading this map is the most straight forward way out. I could add a one-liner to return an immutable map. `def configKeysMap = serverDefaultConfigNames.toMap`